packages/compass-query-bar/src/components/query-bar.tsx

```diff
@@ -0,0 +1,214 @@
+    <form className={queryBarFormStyles} onSubmit={onFormSubmit} noValidate>
+      <div className={queryBarFirstRowStyles}>
+        {showQueryHistoryButton && (
+          <button
+            data-testid="query-history-button"
+            onClick={toggleQueryHistory}
+            className={openQueryHistoryStyles}
+            id="open-query-history"
+            aria-label="Open query history"
+            type="button"
+          >
+            <Icon glyph="Clock" />
+            <Icon glyph="CaretDown" />
+          </button>
+        )}
+        <div className={filterContainerStyles}>
+          <QueryOptionComponent
+            hasError={!queryOptionProps.filterValid}
+            queryOption="filter"
+            onChange={(value: string) => onChangeQueryOption('filter', value)}
+            onApply={onApply}
+            placeholder={
+              queryOptionProps.filterPlaceholder ||
+              OPTION_DEFINITION.filter.placeholder
+            }
+            refreshEditorAction={refreshEditorAction}
+            schemaFields={schemaFields}
+            serverVersion={serverVersion}
+            value={queryOptionProps.filterString}
+          />
+        </div>
+        <Button
+          aria-label="Reset query"
+          data-testid="query-bar-reset-filter-button"
+          onClick={onReset}
+          disabled={queryState !== 'apply'}
+          size="small"
+          type="button"
+        >
+          Reset
+        </Button>
+        <Button
+          data-testid="query-bar-apply-filter-button"
+          disabled={!isQueryValid}
+          variant="primary"
+          size="small"
+          type="submit"
+        >
+          {buttonLabel}
+        </Button>
+        {showExportToLanguageButton && (
+          <IconButton
+            onClick={onOpenExportToLanguage}
+            title="Open export to language"
+            aria-label="Open export to language"
+            data-testid="query-bar-open-export-to-language-button"
+            type="button"
+          >
+            <Icon glyph="Export" />
+          </IconButton>
+        )}
+
+        {queryOptions && queryOptions.length > 0 && (
+          <MoreOptionsToggle
+            aria-controls="additional-query-options-container"
+            data-testid="query-bar-options-toggle"
+            isExpanded={isQueryOptionsExpanded}
+            onToggleOptions={toggleExpandQueryOptions}
+          />
+        )}
+      </div>
+      {queryOptions && queryOptions.length > 0 && (
+        <div id="additional-query-options-container">
+          {isQueryOptionsExpanded && (
+            <QueryOptionsGrid
+              queryOptions={queryOptions}
+              queryOptionProps={queryOptionProps}
+              onChangeQueryOption={onChangeQueryOption}
+              onApply={onApply}
+              refreshEditorAction={refreshEditorAction}
+              schemaFields={schemaFields}
+              serverVersion={serverVersion}
+            />
+          )}
```

Comment:
I think the render lag on mount was significant enough that we didn't want to add them to the view before because it caused too much lag on showing the collection tabs initially. We could get fancier here and maybe render them incrementally or something, but I think we'd probably find better performance improvements in changing our base ace editor.
We aren't addressing that issue with this work, but definitely a good thing to bring up and keep in mind for future.